inc/zoo/swar/SWAR.h

```diff
@@ -472,13 +481,59 @@ booleans(SWAR<NB, T> arg) noexcept {
 template<int NBits, typename T>
 constexpr auto
 differents(SWAR<NBits, T> a1, SWAR<NBits, T> a2) {
-    return booleans(a1 ^ a2);
+    return ~equals(a1, a2);
+}
+
+/**
+ * @return BooleanSWAR that contains a true value in each lane where the two
+ * input SWARs match.
+ */ 
+template<int NBits, typename T>
+constexpr auto
+equals(SWAR<NBits, T> a1, SWAR<NBits, T> a2) noexcept {
+    // Knuth, TAOCP 4A pg 152
+    using S = swar::SWAR<NBits, T>;
+    constexpr auto TAOCP_H{S{S::MostSignificantBit}}, TAOCP_L{S{S::LeastSignificantBit}};
+    auto
+        nullLaneIfEqual = a1 ^ a2,
+        allowSubtractionByOneTurningOnMSB = nullLaneIfEqual | TAOCP_H,
+        nullDetectorViaFlippingMSB_low = allowSubtractionByOneTurningOnMSB - TAOCP_L,
+        equalLanesHaveMSB_off = nullLaneIfEqual | nullDetectorViaFlippingMSB_low,
+        flipMSB = ~equalLanesHaveMSB_off,
+        rv = TAOCP_H & flipMSB;
+    return rv;
```

Comment:
This returns a `SWAR`, not a `BooleanSWAR`.

Reply:
The conversion to `BooleanSWAR` ought to be made in the "public", not be an implementation detail that would require being a friend of `BooleanSWAR`.
`return booleans(rv)` suffices.

Reply:
`booleans` is implemented with a call to `constantIsGreaterEqual`, so it would not suffice.
Previously, I suggested the extra op to convert from a SWAR to a BooleanSWAR could be elided in some cases by not forcing the return immediately to a boolean.